dataprofiler/profilers/profile_builder.py

```diff
@@ -1372,17 +1367,26 @@ def _get_correlation(self, clean_samples):
         :type clean_samples: dict()
         """
         columns = self.options.correlation.columns
+        clean_column_ids = []
         if columns is None:
-            for col in self._profile:
+            for col_id, col in enumerate(self._profile.keys()):
                 if (self._profile[col].profile['data_type'] not in ['int', 'float']
                         or self._profile[col].null_count > 0):
                     clean_samples.pop(col)
+                else:
+                    clean_column_ids.append(col_id)
         if len(clean_samples) <= 1:
             return None
 
         data = pd.DataFrame(clean_samples)
         data = data.apply(pd.to_numeric, errors='coerce')
-        return data.corr()
+
+        n_cols = len(self._profile)
+        corr_mat = np.empty((n_cols, n_cols))
+        corr_mat.fill(np.nan)
+        rows = [[id] for id in clean_column_ids]
+        corr_mat[rows, clean_column_ids] = np.corrcoef(data, rowvar=False)
```

Comment:
i believe we can do
`corr_mat[np.ix_(clean_column_ids, clean_column_ids)] = np.corrcoef(data, rowvar=False)`

Reply:
However, not sure that it is cleaner.

Reply:
I keep it as is for now, at least it's a bit faster than np.ix